SerialPrograms/Source/Controllers/KeyboardInput/KeyboardInput.h

```diff
@@ -122,11 +126,13 @@ class KeyboardManager : public KeyboardInputController{
         deltas.to_state(static_cast<StateType&>(state));
     }
     virtual void cancel_all_commands() override{
+        WallClock time_stamp = current_time();
         WriteSpinLock lg(m_lock);
         if (m_controller == nullptr){
             return;
         }
         m_controller->cancel_all_commands();

```

Comment:
Move the timestamp read to just before this line since it's closer to the actual time being sent to the Switch. Having it before the lock could mean an arbitrary delay.